src/lib/profile.ts

```diff
@@ -404,16 +403,16 @@ export class Profile {
 
   // Demangle symbols for readability
   async demangle() {
-    let demangleCpp: ((name: string) => string) | null = null
+    let demangle: ((name: string) => string) | null = null
 
     for (let frame of this.frames) {
-      // This function converts a mangled C++ name such as "__ZNK7Support6ColorFeqERKS0_"
-      // into a human-readable symbol (in this case "Support::ColorF::==(Support::ColorF&)")
-      if (frame.name.startsWith('__Z')) {
-        if (!demangleCpp) {
-          demangleCpp = (await demangleCppModule).demangleCpp
+      // This function converts a mangled C++ and Rust name into a human-readable symbol.
+      if (frame.name.startsWith('__Z') || frame.name.startsWith('_R') || frame.name.startsWith('_Z')) {
```

Comment:
Hmmm is there any other way of sanely auto-detecting this? `__Z` is so rare as a prefix it seems unlikely to yield false positives, but `_R` and `_Z` seem far more likely and I'd like to avoid incurring this loading cost unless needed

Reply:
I don't see any other way sadly. 

FWIW, _Z is the default mangling prefix and __Z happens only on macOS where symbols get an extra _.

I tested with a raw undemangled C++ perf trace on Linux and speedscope didn't demsngle it. I added this as a result but should have made a separate commit to explain.